### ISSUE TITLE
Exporter fails when /metrics call to Elasticsearch cluster return status 404

### ISSUE THREAD

**Issue by amankedia** [justwatchcom/elasticsearch_exporter#260]
The Elasticsearch exporter fails when Elasticsearch Cluster returns a Status 404

`curl -X GET elasticsearch:9200/metrics
{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index","index_uuid":"_na_","resource.type":"index_or_alias","resource.id":"metrics","index":"metrics"}],"type":"index_not_found_exception","reason":"no such index","index_uuid":"_na_","resource.type":"index_or_alias","resource.id":"metrics","index":"metrics"},"status":404}`

When this exception is returned from the ES cluster, the exporter itself fails.

Logs from Exporter pod
Warning  Unhealthy  7m (x124 over 2h)  kubelet,  IP Readiness probe failed: Get http://IP:9211/metrics: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
  Warning  BackOff    2m (x576 over 2h)  kubelet, IP  Back-off restarting failed container

**Comment by zwopir** [comment]
you need to query `/metrics` on the exporter (port 9114 on latest RC), not on the elasticsearch node. Unless you have an index called `metrics` the above curl doesn't make any sense.

The logs indicate that you've probably misconfigured the health check in the k8s deployment

**Comment by amankedia** [comment]
True. I got inside the Elasticsearch exporter pod. I have modified the ports. I have 2 Exporters, the one on port 9210 works completely fine and its ES cluster is also doing well.
The one on port 9211 is giving the following result on CURL queries and its corresponding ES Cluster is also having issues. I am not sure if there is any relation between the ES Cluster failing and the corresponding exporter failing.
[root@elasticsearch-exporter-6cfcf67858-72xqf /]# curl -X GET elasticsearch-exporter-6cfcf67858-72xqf:9211/metrics
curl: (7) Failed connect to elasticsearch-exporter-6cfcf67858-72xqf:9211; Connection refused
[root@elasticsearch-exporter-6cfcf67858-72xqf /]# curl -X GET elasticsearch-exporter-6cfcf67858-72xqf:9211/       
curl: (7) Failed connect to elasticsearch-exporter-6cfcf67858-72xqf:9211; Connection refused
[root@elasticsearch-exporter-6cfcf67858-72xqf /]# curl -X GET elasticsearch-exporter-6cfcf67858-72xqf:9211 
curl: (7) Failed connect to elasticsearch-exporter-6cfcf67858-72xqf:9211; Connection refused
